activation/poetdb.go

```diff
@@ -0,0 +1,129 @@
+package activation
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	xdr "github.com/nullstyle/go-xdr/xdr3"
+	"github.com/spacemeshos/go-spacemesh/common"
+	"github.com/spacemeshos/go-spacemesh/database"
+	"github.com/spacemeshos/go-spacemesh/types"
+	"github.com/spacemeshos/merkle-tree"
+	"github.com/spacemeshos/poet/hash"
+	"github.com/spacemeshos/poet/shared"
+	"github.com/spacemeshos/poet/verifier"
+	"github.com/spacemeshos/sha256-simd"
+)
+
+type PoetDb struct {
+	store database.DB
+}
+
+func NewPoetDb(store database.DB) *PoetDb {
+	return &PoetDb{store: store}
+}
+
+func (db *PoetDb) AddMembershipProof(proof types.PoetMembershipProof) error {
+	root, err := calcRoot(proof.Members)
+	if err != nil {
+		return fmt.Errorf("failed to calculate membership root: %v", err)
+	}
+
+	var members bytes.Buffer
+	if _, err := xdr.Marshal(&members, proof.Members); err != nil {
+		return fmt.Errorf("failed to marshal members: %v", err)
+	}
+
+	if err := db.store.Put(root, members.Bytes()); err != nil {
+		return fmt.Errorf("failed to store membership proof: %v", err)
+	}
+
+	return nil
+}
+
+func (db *PoetDb) AddPoetProof(proof types.PoetProof) error {
+	if _, err := db.store.Get(proof.MembershipRoot); err != nil {
+		return fmt.Errorf("failed to fetch matching membership proof: %v", err)
```

Comment:
I've considered it, but since this is just a slice of bytes it has no meaning unless it's also printed somewhere else (e.g. when it's created). Didn't think it was worth it - we can always add (to all relevant places) it if we're debugging something. Do you agree?

Reply:
I assumed that it was also printed in the poet logs. Printing it here will help us associate the error with the poet execution. Up to you.

Reply:
well, this particular error can't occur anymore since we merged the messages, but I enriched all the error messages with more data.